core/src/main/java/io/grpc/internal/ServerImpl.java

```diff
@@ -584,16 +584,21 @@ public void runInContext() {
 
     @Override
     public void closed(final Status status) {
+      // For cancellations, promptly inform any users of the context that their work should be
+      // aborted. Otherwise, we can wait until pending work is done.
+      if (!status.isOk()) {
+        context.cancel(status.getCause());
```

Comment:
Isn't the user supplied work `getListener().closed(status)` that is enqueued **after** what we can enqueue here?